text-utils/more.c

```diff
@@ -193,54 +195,77 @@ struct more_control {
-	fputs(_(" -p          do not scroll, clean screen and display text\n"), out);
-	fputs(_(" -s          squeeze multiple blank lines into one\n"), out);
-	fputs(_(" -<number>   the number of lines per screenful\n"), out);
-	fputs(_(" +<number>   display file beginning from line number\n"), out);
-	fputs(_(" +/<string>  display file beginning from search string match\n"), out);
-
-	fputs(USAGE_SEPARATOR, out);
-	printf( "     --help     %s\n", USAGE_OPTSTR_HELP);
-	printf( " -V, --version  %s\n", USAGE_OPTSTR_VERSION);
+	printf("%s", USAGE_HEADER);
+	printf(_(" %s [options] <file>...\n"), program_invocation_short_name);
+
+	printf("%s", USAGE_SEPARATOR);
+	printf("%s\n", _("A file perusal filter for CRT viewing."));
+
+	printf("%s", USAGE_OPTIONS);
+	printf("%s\n", _(" -d, --silent          display help instead of ringing bell"));
+	printf("%s\n", _(" -f, --logical         count logical rather than screen lines"));
+	printf("%s\n", _(" -l, --no-pause        suppress pause after form feed"));
+	printf("%s\n", _(" -c, --print-over      do not scroll, display text and clean line ends"));
+	printf("%s\n", _(" -p, --clean-print     do not scroll, clean screen and display text"));
+	printf("%s\n", _(" -s, --squeeze         squeeze multiple blank lines into one"));
+	printf("%s\n", _(" -u, --plain           suppress underlining and bold"));
+	printf("%s\n", _(" -n, --lines <number>  the number of lines per screenful"));
+	printf("%s\n", _(" -<number>             same as --lines"));
+	printf("%s\n", _(" +<number>             display file beginning from line number"));
+	printf("%s\n", _(" +/<pattern>           display file beginning from pattern match"));
+	printf("%s", USAGE_SEPARATOR);
+	printf(USAGE_HELP_OPTIONS(23));
 	printf(USAGE_MAN_TAIL("more(1)"));
 	exit(EXIT_SUCCESS);
 }
 
-
-static void arg_parser(struct more_control *ctl, char *s)
+static void argscan(struct more_control *ctl, int as_argc, char **as_argv)
 {
-	int seen_num = 0;
-
-	while (*s != '\0') {
-		switch (*s) {
-		case '0':
-		case '1':
-		case '2':
-		case '3':
-		case '4':
-		case '5':
-		case '6':
-		case '7':
-		case '8':
-		case '9':
-			if (!seen_num) {
-				ctl->lines_per_screen = 0;
-				seen_num = 1;
+	int c, opt;
+	static const struct option longopts[] = {
+		{ "silent",      no_argument,       NULL, 'd' },
+		{ "logical",     no_argument,       NULL, 'f' },
+		{ "no-pause",    no_argument,       NULL, 'l' },
+		{ "print-over",  no_argument,       NULL, 'c' },
+		{ "clean-print", no_argument,       NULL, 'p' },
+		{ "squeeze",     no_argument,       NULL, 's' },
+		{ "plain",       no_argument,       NULL, 'u' },
+		{ "lines",       required_argument, NULL, 'n' },
+		{ "version",     no_argument,       NULL, 'V' },
+		{ "help",        no_argument,       NULL, 'h' },
+		{ NULL, 0, NULL, 0 }
+	};
+
+	/* Take care of number option and +args. */
+	for (opt = 0; opt < as_argc; opt++) {
+		int move = 0;
+
+		if (as_argv[opt][0] == '-' && isdigit(as_argv[opt][1])) {
```

Comment:
Maybe use isdigit_string() instead of isdigit()

Reply:
Fixed to use isdigit_string() in ffb1f57ee2931f019e6cf72b121bd66892b34fa7.